### ISSUE TITLE
Migrate password hashing functions (PBKDF2, Scrypt, etc..) to Argon2id

### ISSUE THREAD

**Issue by RichardFevrier** [cozy/cozy-stack#2519]
Hello everybody 👋 thank you for your amazing work !

I was wondering now that the hashing function Argon2id is available in the official [crypto module](https://pkg.go.dev/golang.org/x/crypto/argon2?tab=doc) if it was not the time to migrate the current Cozy's password hashing stack ?

**Comment by nono** [comment]
I don't think so. Argon2id is known as complicated to correctly tune for the environment ([from OWASP](https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html#argon2id)) and the current password hashing function is not deprecated or with known vulnerabilities. It would take time to write a migration to Argon2id, and I really think that this time is better use for other purpose.
